rollbar-web/src/main/java/com/rollbar/web/listener/RollbarRequestListener.java

```diff
@@ -0,0 +1,26 @@
+package com.rollbar.web.listener;
+
+import javax.servlet.ServletRequestEvent;
+import javax.servlet.ServletRequestListener;
+import javax.servlet.http.HttpServletRequest;
+
+public class RollbarRequestListener implements ServletRequestListener {
+
+  private static final ThreadLocal<HttpServletRequest> CURRENT_REQUEST = new ThreadLocal<>();
```

Comment:
Are we guaranteed to have a single request per thread? If not, another way of implementing this which I think I prefer, is to attach a new instance of the Rollbar notifier to each request and expose it to the user somehow.

Reply:
Yes, AFAIK every request in a servlet container is manage by a different thread. So, if I am not wrong in a servlet container when a request reaches it, it will assign a thread from its "requests" thread pool to handle it.